### ISSUE TITLE
Fix fluids not having bottom texture planes on the inside

### ISSUE THREAD

**Issue by muzikbike** [jellysquid3/Sodium#1210]
### Version information

sodium-fabric-mc1.18.2-0.4.1+build.15

### Expected Behavior

Since the other five faces exist, the bottom should as well.

### Actual Behavior

It does not under any circumstances.

### Reproduction Steps

1. Place water above a pressure plate, with the water surrounded by four top slabs for easy access
2. Enter the water

### Java version

17.0.2 64bit

### CPU

4× Intel® Core™ i3-3240 CPU @ 3.40GHz

### GPU

Intel® HD Graphics 2500

### Additional information

Vanilla bug https://bugs.mojang.com/browse/MC-190053, which has persisted since 1.7. Versions prior to 1.7 did not have this bug.

This has been split from #91 due to changes made to fluid rendering in 1.18.2.
![2022-03-05_00 40 20](https://user-images.githubusercontent.com/52297970/156860361-11e2fbaf-484b-41b8-b5a6-f451ce30f8aa.png)

**Comment by MeeniMc** [comment]
0.4.10 looks like Vanilla (i.e., it is missing the bottom texture). This is a vanilla issue and a request for enhancement, not a bug.

**Comment by muzikbike** [comment]
I fail to see how this is fixed in the latest version of Sodium for 1.20?
![2023-06-19_09 58 07](https://github.com/CaffeineMC/sodium-fabric/assets/52297970/d81ab0b8-51e8-42d0-91b1-c51032a35ac5)

**Comment by pajicadvance** [comment]
You're right, I closed this thinking it was related to the water overlay texture bug.

**Comment by sedyh** [comment]
Hello. Not sure if it is a spam. But I still have gaps at still water (any degree) on 1.19.2 + sodium 0.4.4+build.18.
The last case I met was [#1473652041](https://github.com/CaffeineMC/sodium-fabric/issues/1489#issuecomment-1473652041) (year ago).
I can't reply in #38 since it was blocked.

**Comment by douira** [comment]
This is not the same as #38. If you are experiencing #38, you don't need to add reports of it since it's known that it exists (and will continue to until it is deliberately fixed).

**Comment by sedyh** [comment]
> This is not the same as #38.

Oh, looks like it's a bit different. Should I post a new issue then?

**Comment by douira** [comment]
If you are experiencing #1489 no further action is needed since it's the same as #38. If you are experiencing this issue (missing bottoms of water) also nothing needs to be done since it's a vanilla issue (and this issue already exists). If you are experiencing something that we don't already have an open issue on, you can make a new issue.

**Comment by sedyh** [comment]
I think I'm seeing chunk borders under the ice which is mentioned in [#659070443](https://github.com/CaffeineMC/sodium-fabric/issues/38#issuecomment-659070443). This issue only happens when sodium installed on any version after 1.18.2. I don't think it's connected to vanilla.

**Comment by douira** [comment]
Then you are seeing an instance of #38. You don't need to make a bug report about this since it's well-known.

**Comment by douira** [comment]
I believe the effect of a fix for this issue can be simulated by removing ` || !this.isSideExposed(level, posX, posY, posZ, Direction.DOWN, 0.8888889F)` from [this line](https://github.com/CaffeineMC/sodium-fabric/blob/d2a68153e5bb6ad59258db5b8e3604e8292f565b/src/main/java/net/caffeinemc/mods/sodium/client/render/chunk/compile/pipeline/DefaultFluidRenderer.java#L114). Here's a picture of it before and after this change:
![2024-05-05_19 26 12](https://github.com/CaffeineMC/sodium-fabric/assets/17120022/2f67a5c9-708f-4e35-9f58-5576ce39a2de)
![2024-05-05_19 28 19](https://github.com/CaffeineMC/sodium-fabric/assets/17120022/b6dc0188-e800-4179-a1a0-8782e491243b)
As you can see it's much darker overall and very different looking. I also think that for consistency reasons you'd need to apply the same effect to the sides of blocks, which makes it even weirder. IMO this is an art direction question and not simply an issue of "the fluid renderer" is wrong. Even if it could be construed as a matter of visual consistency, it's anything but consistent with what I'd expect to see on the inside of a fluid. (Assuming this patch is actually correct)
Here's another example of the radical difference, under water in a river:
![2024-05-05_19 31 27](https://github.com/CaffeineMC/sodium-fabric/assets/17120022/045b9bef-55c9-4a63-a1d1-d2d961c6a710)
![2024-05-05_19 31 48](https://github.com/CaffeineMC/sodium-fabric/assets/17120022/19743cb9-06c7-4d06-af40-dcab514a73d3)
You can see there's some lighting issues here that would need to be addressed but I think it gets the point across. It's obvious that the sides of blocks would also need to be affected. Possibly making all the inner water surfaces underwater full brightness would make the whole scene less dark, but it would certainly be very blue. Realistically, it also doesn't make much sense to have inner water surfaces against blocks, because you know, that's not how water surfaces work irl either.

As mentioned, IMO this is a question of art direction and should be answered by vanilla. I'm going to post my images on the vanilla bug too.

**Comment by muzikbike** [comment]
Why isn't the bottom face culled against solid blocks in the screenshots? The face should only be rendered if there's no full block below (check vanilla 1.6.4 for the expected behaviour).

**Comment by douira** [comment]
ignore me, I just completely misunderstood what this was saying.

**Comment by muzikbike** [comment]
No,  you were almost right: the interior bottom face should only cull against opaque blocks. It should render as it did if there's air, glass or some other non-opaque block below.

Screenshot from 1.6.4, displaying expected behaviour (bottom face is visible against glass, but not against stone):
![2024-05-05_20 22 54](https://github.com/CaffeineMC/sodium-fabric/assets/52297970/09e5eb53-3a81-4beb-b955-2bbc457115d4)

Screenshot from 1.7.9, displaying current undesirable behaviour (bottom face is never visible):
![2024-05-05_20 23 56](https://github.com/CaffeineMC/sodium-fabric/assets/52297970/9e5b97e6-8b88-49a7-b6b4-97a5d8d2835e)
